### ISSUE TITLE
Shift key not working on keyboard

### ISSUE THREAD

**Issue by velardevines** [vmihalachi/turbo-editor#107]
Shift key on Microsoft keyboard is not working. Using Nexus 7 2013 with Microsoft Universal Mobile Keyboard.

**Comment by softfoot** [comment]
I have the same problem with a bluetooth keyboard, a simple USB keyboard and a Logitec USB keyboard.

They all work OK with all other apps.

Dave

**Comment by kracejic** [comment]
Same problem is on Asus Transformer TF701t.

**Comment by z3t0** [comment]
Try the different keyboard layouts in this app. https://play.google.com/store/apps/details?id=com.medion.android.keyboard&hl=en 
The problem may not be turbo, but rather the keyboard layout set. Does shift work outside of turbo?

**Comment by velardevines** [comment]
Yes, Shift key works on all other apps. turbo is the only app in which it
does not work.

--Bill

**Comment by kracejic** [comment]
Yes, the problem manifests itself only on turbo.

**Comment by Theodus2009** [comment]
I have experienced the same issue on my Xperia Z3 running Android 5.1.1.

**Comment by Yajo** [comment]
Same here in Samsung Galaxy Tab A (see those upper case letters I wrote? Not possible in app).

**Comment by duggum** [comment]
Same problem with version 2.4 using two different bluetooth keyboards and a usb keyboard connected via OTG. All keyboards work fine in other apps.

Using Samsung NotePro 12.2

**Comment by ZakariaJawas** [comment]
I have the same problem with shift key
Im using a bluetooth keyboard it works fine in other appa except in turbo editor
Im using Lg gpad

**Comment by placeofexit** [comment]
Same problem here: bluetooth iWerkz keyboard on Samsun Galaxy s5, works fine in other apps.

**Comment by dawaltconley** [comment]
Same problem here: bluetooth iWerkz keyboard on Samsun Galaxy s5, works fine in other apps.

**Comment by KaisarCode** [comment]
Same problem here. Bluetooth keyboard brand Genius.

**Comment by lucilio** [comment]
Same problem: Asus Vivobook x202e running Android-X86-64.

**Comment by lordmirg** [comment]
I have the same issue on the Pixel C (with UK keyboard layout) Keyboard.  Shift works in other apps (I'm typing this on my Pixel C right now).

**Comment by JacobWardell** [comment]
I have a Samsung Galaxy S5 Android phone, and I tried a lot of stuff to get the Shift Key to work on my USB keyboard when I had it plugged into my phone.  I found that the stock Google/Android keyboard app, the Samsung Keyboard app, and the SwiftKey Keyboard app did not solve my problem.  I did come across an app however that DOES solve the Shift key problem:  The "Go Keyboard" app:  https://play.google.com/store/apps/details?id=com.jb.emoji.gokeyboard&hl=en

After I installed this app and set it as the default keyboard app on my Android phone (Samsung Galaxy S5), the Shift key worked just fine on my phone (capitalizing letters and allowing me to type symbols above the number keys). 

I hope this works for you, too!

**Comment by lordmirg** [comment]
Go keyboard doesn't sound like the sort of thing I want to install on my device.   Interestingly though, the Google Japanese Input keyboard allows me to type capitals.

**Comment by markqvist** [comment]
Insane this hasn't been fixed after almost 3 years. Just bought the pro version only to find out about this afterwards. Editor that doesn't support shift key is a bit pointless. Wondering if developer has abandoned the app? Seriously considering forking, fixing and re-releasing for free with no paid upgrade...

**Comment by adrianmalacoda** [comment]
Hi @markqvist 

Unfortunately it seems that the developer has abandoned this project. I started work on a fork a couple years ago and unfortunately I have also let it lapse (other things come up...). I have a few things I want to look at before I make a release.

In the mean time though, if you're interested here is the fix for hardware keyboards. https://github.com/adrianmalacoda/text-editor-8000/commit/46ad99aa464251995141613ba89abe6b4623b26c

I did not submit this as a PR to this project because it is abandoned.

**Comment by markqvist** [comment]
Thanks a lot @adrianmalacoda! I'll clone you repo and build your version :)
